modules/planning/tasks/deciders/st_bounds_decider/st_obstacles_processor.cc

```diff
@@ -42,109 +43,148 @@ using apollo::common::PathPoint;
     PathDecision* const path_decision) {
   // Sanity checks.
   CHECK_NOTNULL(path_decision);
+  CHECK_GT(planning_time_, 0.0);
+  CHECK_GT(planning_distance_, 0.0);
+  CHECK_GT(path_data_.discretized_path().size(), 1);
 
   // Go through every obstacle.
-
   return Status::OK();
 }
 
 // TODO(jiacheng): implement this.
-std::pair<double, double> STObstaclesProcessor::GetRegularBoundaryFromObstacles(
-    double t) {
-  return 0.0;
+std::pair<double, double>
+STObstaclesProcessor::GetRegularBoundaryFromObstacles(double t) {
+  return {0.0, 0.0};
 }
 
 // TODO(jiacheng): implement this.
 std::pair<double, double>
 STObstaclesProcessor::GetFallbackBoundaryFromObstacles(double t) {
-  return 0.0;
+  return {0.0, 0.0};
 }
 
 ///////////////////////////////////////////////////////////////////////////////
 // Private helper functions.
 
 bool STObstaclesProcessor::ComputeObstacleSTBoundary(const Obstacle& obstacle) {
-  std::vector<STPoint> lower_points;
-  std::vector<STPoint> upper_points;
-  const auto& adc_path_points = path_data_.discretized_path();
-  const auto& obs_trajectory = obstacle.Trajectory();
-
-  if (obs_trajectory.trajectory_point_size() == 0) {
-    // Processing a static obstacle.
-    // Sanity checks.
-    if (!obstacle.IsStatic()) {
-      AWARN << "Non-static obstacle[" << obstacle.Id()
-            << "] has NO prediction trajectory."
-            << obstacle.Perception().ShortDebugString();
-    }
-  } else {
-    // Processing a dynamic obstacle.
-  }
+  // std::vector<STPoint> lower_points;
+  // std::vector<STPoint> upper_points;
+  // const auto& adc_path_points = path_data_.discretized_path();
+  // const auto& obs_trajectory = obstacle.Trajectory();
+
+  // if (obs_trajectory.trajectory_point_size() == 0) {
+  //   // Processing a static obstacle.
+  //   // Sanity checks.
+  //   if (!obstacle.IsStatic()) {
+  //     AWARN << "Non-static obstacle[" << obstacle.Id()
+  //           << "] has NO prediction trajectory."
+  //           << obstacle.Perception().ShortDebugString();
+  //   }
+  // } else {
+  //   // Processing a dynamic obstacle.
+  // }
   return true;
 }
 
 bool STObstaclesProcessor::GetOverlappingS(
     const std::vector<PathPoint>& adc_path_points,
     const Box2d& obstacle_instance, const double adc_l_buffer,
     std::pair<double, double>* const overlapping_s) {
+  // Locate the possible range to search in details.
+  int pt_before_idx = GetSBoundingPathPointIndex(
+      adc_path_points, obstacle_instance,
+      vehicle_param_.front_edge_to_center(),
+      true, 0, static_cast<int>(adc_path_points.size()) - 2);
+  int pt_after_idx = GetSBoundingPathPointIndex(
+      adc_path_points, obstacle_instance,
+      vehicle_param_.back_edge_to_center(),
+      false, 0, static_cast<int>(adc_path_points.size()) - 2);
+  if (pt_before_idx == static_cast<int>(adc_path_points.size()) - 2)
+    return false;
+  if (pt_after_idx == 0)
+    return false;
```

Comment:
Always add brackets for `if`, `for`, `while`.

Reply:
Done.